src/test/recovery/t/139_archive_mode_always.pl

```diff
@@ -0,0 +1,82 @@
+# test wal persistance for archive_mode 'always'
+use strict;
+use warnings;
+use PostgresNode;
+use TestLib;
+use Test::More tests => 2;
+use File::Copy;
+
+# The idea of the test is to check that not archived WAL files are not removed
+# by creation of checkpoint (on master) or restartpoint (on standby).
+
+# Initialize master node, doing archives
+my $node_master = get_new_node('master');
+$node_master->init(
+	has_archiving    => 1,
+	allows_streaming => 1);
+my $backup_name = 'my_backup';
+my $master_connstr = $node_master->connstr;
+
+# Append necessary config. Some of params override already initialized ones.
+# archive_mode 'always' is a subject of the test.
+# archive_command 'false' is used to prevent archiving. The idea is to check
+#   WAL file is not removed by checkpointer. Successfull archiving (with file
+#   removing) should not affect the test.
+# wal_keep_segments '0' is used to prevent keeping old WAL files. As test
+#   checks that WAL file still exist, it should not be kept by any other
+#   mechanics.
+# create_restartpoint_on_ckpt_record_replay 'on' makes restartpoint (and
+#   waiting for completion) while replaying checkpoint record on standby.
+$node_master->append_conf(
+		'postgresql.conf', qq{
+archive_mode = 'always'
+archive_command = '/usr/bin/false'
+wal_keep_segments = 0
+create_restartpoint_on_ckpt_record_replay = 'on'
+});
+
+# Start it
+$node_master->start;
+
+# Take backup for standby
+$node_master->backup($backup_name);
+
+# Initialize standby node from backup, fetching WAL from archives
```

Comment:
The same as above - "Mixing continious and non-continious forms of verbs"

Reply:
The same as above.